python/lib/communication/dmod/communication/client.py

```diff
@@ -252,147 +383,243 @@ def _check_if_new_session_needed(self, use_current_values: bool = True) -> bool:
-                path = '/' + path
-        return '{}://{}:{}{}'.format(proto, host.strip(), str(port).strip(), path)
-
-    def __init__(self, ssl_directory: Path, *args, **kwargs):
+    def __init__(self, transport_client: TransportLayerClient, default_response_type: Optional[Type[Response]] = None,
+                 *args, **kwargs):
         """
-        Initialize this instance.
+        Initialize.
 
         Parameters
         ----------
-        ssl_directory
+        transport_client : TransportLayerClient
+            The client for handling the underlying raw OSI transport layer communications with the service.
+        default_response_type: Optional[Type[Response]]
+            Optional class type for responses, to use when no response class param is given when making a request.
         args
         kwargs
+        """
+        self._transport_client = transport_client
+        self._default_response_type: Optional[Type[Response]] = default_response_type
 
-        Other Parameters
+    def _process_request_response(self, response_str: str, response_type: Optional[Type[Response]] = None) -> Response:
+        """
+        Process the serial form of a response returned by ::method:`async_send` into a response object.
+
+        Parameters
         ----------
-        endpoint_uri : str
-            The endpoint for the client to connect to when opening a connection, for superclass init.
+        response_str : str
+            The string returned by a request made via ::method:`async_send`.
+        response_type: Optional[Type[Response]]
+            An optional class type for the response that, if ``None`` (the default) is replaced with the default
+            provided at initialization.
+
+        Returns
+        -------
+        Response
+            The inflated response object.
+
+        See Also
+        -------
+        async_send
         """
-        super().__init__(*args, **kwargs)
+        if response_type is None:
+            response_type = self._default_response_type
 
-        self._ssl_directory = ssl_directory
-        """Path: The parent directory of the cert PEM file used for the client SSL context."""
+        response_json = {}
+        try:
+            # Consume the response confirmation by deserializing first to JSON, then from this to a response object
+            response_json = json.loads(response_str)
+            try:
+                response_object = response_type.factory_init_from_deserialized_json(response_json)
+                if response_object is None:
+                    msg = f'********** {self.__class__.__name__} could not deserialize {response_type.__name__} ' \
+                          f'from raw websocket response: `{str(response_str)}`'
+                    reason = f'{self.__class__.__name__} Could Not Deserialize To {response_type.__name__}'
+                    response_object = response_type(success=False, reason=reason, message=msg, data=response_json)
```

Comment:
> And then we'd need to handle the exception properly somewhere else further up the stack.

Right, this is my primary motivation for trying to treat the exceptions as values instead of throwing here. Treating them as values without attaching for example, a traceback, could also become difficult to debug. I personally think it is worth the headache though. Exceptions are so easy to ignore / mishandle.



> Regardless, addressing this properly would be fairly intricate, and what's in place now was carried over from previous code, albeit perhaps arranged slightly differently. I.e., it's out of scope for this PR, but it deserves its own issue.

Also totally fair, thanks for calling out of the scope creep!

Reply:
FYI, I've opened #418 for this.